src/flow/assembly_lmh_old.hh

```diff
@@ -28,598 +47,688 @@
-            double bc_flux = -af_->bc_flux.value(b_ele.centre(), b_ele);
-            double bc_pressure = af_->bc_pressure.value(b_ele.centre(), b_ele);
-            double bc_sigma = af_->bc_robin_sigma.value(b_ele.centre(), b_ele);
-            
-            dirichlet_edge[sidx] = 2;  // to be skipped in LMH source assembly
-            loc_system_.add_value(edge_row, edge_row,
-                                    -b_ele.measure() * bc_sigma * cross_section,
-                                    (bc_flux - bc_sigma * bc_pressure) * b_ele.measure() * cross_section);
-        }
-        else if (type==DarcyMH::EqFields::seepage) {
-            ad_->is_linear=false;
+            eq_data_->loc_system_[bulk_local_idx_].add_value(edge_row_, edge_row_,
+                    -b_ele.measure() * eq_fields_->bc_robin_sigma(p_bdr) * cross_section_,
+                    (-eq_fields_->bc_flux(p_bdr) - eq_fields_->bc_robin_sigma(p_bdr) * eq_fields_->bc_pressure(p_bdr)) * b_ele.measure() * cross_section_);
+        } else if (type==DarcyMH::EqFields::seepage) {
+            eq_data_->is_linear=false;
 
-            char & switch_dirichlet = ad_->bc_switch_dirichlet[b_ele.idx()];
-            double bc_pressure = af_->bc_switch_pressure.value(b_ele.centre(), b_ele);
-            double bc_flux = -af_->bc_flux.value(b_ele.centre(), b_ele);
-            double side_flux = bc_flux * b_ele.measure() * cross_section;
+            char & switch_dirichlet = eq_data_->bc_switch_dirichlet[b_ele.idx()];
+            bc_pressure_ = eq_fields_->bc_switch_pressure(p_bdr);
+            side_flux_ = -eq_fields_->bc_flux(p_bdr) * b_ele.measure() * cross_section_;
 
             // ** Update BC type. **
-            if(use_dirichlet_switch){  // skip BC change if only reconstructing the solution
-            if (switch_dirichlet) {
-                // check and possibly switch to flux BC
-                // The switch raise error on the corresponding edge row.
-                // Magnitude of the error is abs(solution_flux - side_flux).
-                
-                // try reconstructing local system for seepage BC
-                load_local_system(side.cell());
-                double solution_flux = reconstructed_solution_[side_row];
-
-                if ( solution_flux < side_flux) {
-                    //DebugOut().fmt("x: {}, to neum, p: {} f: {} -> f: {}\n", b_ele.centre()[0], bc_pressure, solution_flux, side_flux);
-                    switch_dirichlet=0;
+            if (use_dirichlet_switch_) {  // skip BC change if only reconstructing the solution
+                if (switch_dirichlet) {
+                    // check and possibly switch to flux BC
+                    // The switch raise error on the corresponding edge row.
+                    // Magnitude of the error is abs(solution_flux - side_flux).
+
+                    // try reconstructing local system for seepage BC
+                    load_local_system(cell_side.cell());
```

Comment:
first make a pure function to assembly the local system we need here, then make it work with its duplicity cal, then possibly we find a new way how to cache it. We may even try to store all local systems as we now have to reassemble during the reconstruct assembly.